src/OpenTelemetry.Shared/LocalFileBlob.cs

```diff
@@ -0,0 +1,121 @@
+// <copyright file="LocalFileBlob.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+
+namespace OpenTelemetry.Shared
+{
+    /// <summary>
+    /// The <see cref="LocalFileBlob"/> allows to save a blob
+    /// in file storage.
+    /// </summary>
+    public class LocalFileBlob : IPersistentBlob
```

Comment:
Question: Do we want this to be part of our public API? I'm thinking instead of having the shared dll we add these artifacts to API or SDK as `internal` and then use the compile link trick to expose them in the projects that need them. It's not the most elegant mechanism but it achieves code reuse without making things public 🤷

Reply:
Good question. If exporters outside of Otel repo could take advantage this implementation we should move it to API or SDK. Could you please share your thoughts?

Reply:
So, the end goal for this is to publish this as a nuget as well, is that correct?

Reply:
I think there is a common use for other vendor specific exporters (e.g. GCP, Azure, AWS) to leverage a local persistent storage interface/implementation.
Regarding whether this should be part of the SDK NuGet or not, I think either way would work, I can see potential concern about binary size for folks who don't need this feature, and also a potential debate that this is specific to batch exporters.

Reply:
No strong opinons. Okay to ship it as sep. package.